substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -81,46 +160,126 @@
 
     @Override
     public Stream<DebugDataInfo> dataInfoProvider() {
-        return Stream.empty();
+        return heap.getObjects().stream().filter(this::acceptObjectInfo).map(this::createDebugDataInfo);
     }
 
-    /**
-     * Implementation of the DebugCodeInfo API interface that allows code info to be passed to an
-     * ObjectFile when generation of debug info is enabled.
+    static ObjectLayout OBJECTLAYOUT = ConfigurationValues.getObjectLayout();
```

Comment:
Never store ImageSingletons in static fields.

ImageSingleton values are only valid for a single native-image compilation. If more than one image gets built within the same JVM instance the `OBJECTLAYOUT` value of the first build will be reused for the second build. In this particular instance you can replace the static field with a static method ...
```java
static ObjectLayout getObjectLayout() {
return ConfigurationValues.getObjectLayout();
}
```
... if using `ConfigurationValues.getObjectLayout()` feels too verbose.

Reply:
Done

Reply:
Hmm ... Did you forget to push the related commit/s ?